src/test/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/ciagmigration/mapper/InductionMigrationMapperTest.kt

```diff
@@ -0,0 +1,375 @@
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.resource.model.HopingToWork
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.resource.model.InPrisonTrainingType
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.resource.model.InPrisonWorkType
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.resource.model.PersonalInterest
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.resource.model.PersonalSkill
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.resource.model.ReasonNotToWork
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.resource.model.TrainingType
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.resource.model.WorkInterestDetail
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.resource.model.WorkType
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.resource.model.aValidAchievedQualification
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.resource.model.aValidCiagInductionResponse
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.resource.model.aValidEducationAndQualificationsResponse
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.resource.model.aValidPreviousWorkResponse
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.resource.model.aValidPrisonWorkAndEducationResponse
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.resource.model.aValidSkillsAndInterestsResponse
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.resource.model.aValidWorkExperienceResource
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.resource.model.aValidWorkInterestsResponse
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneOffset.UTC
+import java.time.temporal.ChronoUnit
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.jpa.entity.HighestEducationLevel as HighestEducationLevelEntity
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.jpa.entity.HopingToWork as HopingToWorkEntity
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.jpa.entity.InPrisonTrainingType as InPrisonTrainingTypeEntity
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.jpa.entity.InPrisonWorkType as InPrisonWorkTypeEntity
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.jpa.entity.TrainingType as TrainingTypeEntity
+
+class InductionMigrationMapperTest {
+
+  private val localDateTimeMapper = LocalDateTimeMapper()
+  private val workOnReleaseMapper = WorkOnReleaseMigrationMapperImpl().also {
```

Comment:
Rather than creating a separate unit test for each mapper and then mocking them in this one, I think it's better to test them fully from the top level one. The downside is the injection of the `localDateTimeMapper` in all the mappers here, plus the massive test method below!

However, using mocks means you get to dictate what each mapper returns, which means it's not tested "end to end" (except at the integration test level, which doesn't exist yet).